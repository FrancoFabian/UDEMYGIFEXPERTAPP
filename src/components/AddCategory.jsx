import { useState } from "react"

export const AddCatgory = ({setCategories}) => {
    const [inputValu, setInputValu] = useState('')
    const onInputChange = ({target}) => {
        setInputValu(target.value)
    }
    const onSubmit = (e) => {
       e.preventDefault();
       if(inputValu.trim().length <= 1)return;
       console.log(inputValu)
       setCategories(c => [inputValu,...c])
       setInputValu('');
    }
    return (
        
        <form onSubmit={(e) => onSubmit(e)}>
            <input
                type="text"
                placeholder="Buscar gifs"
                value={inputValu}
                onChange={onInputChange}
            ></input>
        </form>
       
        
    )
}