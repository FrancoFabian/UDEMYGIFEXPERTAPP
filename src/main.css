*{
    margin: 0;
    padding: 0;
    box-sizing: border-box;
    color: aliceblue;
}
body {
    padding: 60px;
    background-color: rgb(30, 30, 39);
  }
  
  input {
    background-color: rgba(255, 255, 255, 0);
    border-radius: 5px;
    border: 1px solid rgb(97, 32, 158);
    color: rgb(226, 248, 254);
    font-size: 1.2rem;
    outline: none;
    padding: 10px 15px;
    width: 100%;
  }
  
  h2 {
    font-size: 1.5rem;
  }
  
  h3 {
    font-size: 3rem;
    margin-bottom: 5px;
  }
  
  .card-grid {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: center;
  }
  
  .card {
    align-content: center;
    align-items: center;
    background-color: rgb(154, 8, 227);
    border-radius: 10px;
    display: flex;
    flex-direction: column;
    height: 0%;  
    justify-content: center;
    margin-bottom: 20px;
    margin-right: 20px;
    overflow: hidden;
  }
  
  .card p {
    flex: 1;
    font-size: 1.5rem;
    margin-top: 5px;
    padding: 5px 20px 0px 20px;
    text-align: center;
  }
  
  .card img {
    width: 100%;
  }