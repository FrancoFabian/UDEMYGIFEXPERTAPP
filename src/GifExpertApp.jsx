import { useState } from "react";
import { AddCatgory } from "./components/AddCategory";

export const GifExpertApp = () => {
    const [categories, setCategories] = useState(['Maxma','Goku','Naruto']);
    
        return (
        <>
        <AddCatgory setCategories={setCategories}/>
        <ol>
            {
                categories.map((catgory,id)=>{
                   return <li key={id}>{catgory}</li>
                })
            }
        </ol>
        </>
    );
}