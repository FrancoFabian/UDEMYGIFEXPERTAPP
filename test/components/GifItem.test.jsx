import { render,screen } from '@testing-library/react'
import { GifItem } from '../../src/components'
import React from 'react'
describe('Pruebas en <GifItem />', () => {
    const title = "One Punch";
    const url = 'http://one-punch.com/saitma.jpg';
    test('Debe de hacer match con el snapshot',() => {
          const { container } = render( <GifItem title={ title } url={ url }/> );
          expect( container ).toMatchSnapshot();
        })

    test('debe mostrar la imagen con la url inidicad', () => {
        render(<GifItem title={ title } url={ url }/>);
        //expect( screen.getByRole('img').src).toBe(url);
        const {src, alt} = screen.getByRole('img')
        expect( src ).toBe( url );
        expect( alt ).toBe( alt );
    });
    test('Debe mostrar el titulo en el componente', () => {
        render(<GifItem title={ title } url={ url }/>);
        expect( screen.getByText( title )).toBeTruthy()
    });
})