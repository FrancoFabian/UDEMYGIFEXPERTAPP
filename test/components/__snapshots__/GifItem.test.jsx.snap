// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="One Punch"
      src="http://one-punch.com/saitma.jpg"
    />
    <p>
      One Punch
    </p>
  </div>
</div>
`;
